File: app/Models/Performance.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Performance extends Model
{
    use HasFactory;

    protected $fillable = ['username_id', 'exercise_id', 'duration', 'speed', 'distance'];

    public function user()
    {
        return $this->belongsTo(User::class, 'username_id');
    }

    public function exercise()
    {
        return $this->belongsTo(Exercise::class);
    }
}

File: lib/models/performance.dart
class Performance {
  final int id;
  final int duration;
  final int? speed;
  final double? distance;

  Performance({
    required this.id,
    required this.duration,
    this.speed,
    this.distance,
  });

  factory Performance.fromJson(Map<String, dynamic> json) {
    return Performance(
      id: json['id'],
      duration: json['duration'],
      speed: json['speed'],
      distance: json['distance']?.toDouble(),
    );
  }
}
